packages/app/src/cli/services/deploy.ts

```diff
@@ -119,6 +125,66 @@ export async function deploy(options: DeployOptions) {
         },
       ]
 
+      if (partnersApp.betas?.declarativeWebhooks) {
+        tasks.push({
+          title: 'Releasing webhooks',
+          task: async () => {
+            if (!('webhooks' in app.configuration)) return
+
+            // normalize webhook config with the top level config
+            const webhookSubscriptions: NormalizedWebhookSubscriptions = []
+            const {topics, subscriptions, ...webhookConfig} = app.configuration.webhooks
+            const topLevelDestination = pickBy(
+              webhookConfig,
+              (value, key) =>
+                ['subscription_endpoint_url', 'arn', 'pubsub_project', 'pubsub_topic'].includes(key) && value,
+            )
+
+            // top level subscriptions
+            if (Object.keys(topLevelDestination).length && topics?.length) {
+              for (const topic of topics) {
+                webhookSubscriptions.push({
+                  topic,
+                  ...topLevelDestination,
+                })
+                // we could add the defaults here as well (include_fields: [], metafield_namespaces: [], format: 'json)
+                // but I think thats better suited for server side / less info over the wire
```

Comment:
Thoughts? I think probably not and I'll remove the comment, we can default on the server

Reply:
I agree to default on the server and remove the comment 👍